stackstate_checks_base/stackstate_checks/base/utils/http_helper.py

```diff
@@ -0,0 +1,653 @@
+        ** Affects: None **
+    """
+    def get_headers(self, session=False):
+        session = self.session_focus if session is False else session
+        if session:
+            return self.session.headers
+        else:
+            return self.request_object.headers
+
+    """
+        Set authentication on the http request. The authentication details is validated against a schematic
+
+        @schematic
+            This is the HTTP Enum that is used to test the details passed down
+        @details
+            A dict of the credentials required in the 'schematic' enum
+        @session
+            Should this be applied on the session on request
+    """
+    def set_auth(self, schematic=None, details=None, session=False):
+        session = self.session_focus if session is False else session
+        # Reset session auth
+        if (schematic is None or schematic.value is HTTPAuthenticationType.NoAuth or schematic.value is None) \
+                and session is True:
+            self.session.auth = None
+
+        # Reset request object auth
+        elif (schematic is None or schematic.value is HTTPAuthenticationType.NoAuth or schematic.value is None) \
+                and session is False:
+            self.request_object.auth = None
+
+        # If the auth method and data is valid
+        elif schematic is not None and \
+                schematic.value in self.available_authentication_types and \
+                isinstance(details, dict):
+            try:
+                # Validate data
+                data = schematic.value(details)
+                data.validate()
+
+                # Apply session data
+                if schematic is HTTPAuthenticationType.BasicAuth and session is True:
+                    self.session.auth = HTTPBasicAuth(data.username, data.password)
+
+                # Apply request data
+                elif schematic is HTTPAuthenticationType.BasicAuth and session is False:
+                    self.request_object.auth = HTTPBasicAuth(data.username, data.password)
+
+            # If validation failed
+            except DataError:
+                self.handle_error("Auth details does not match the schematic")
+                if session is True:
+                    self.session.auth = None
+                else:
+                    self.request_object.auth = None
+        else:
+            self.handle_error("Invalid authentication method supplied")
+
+    """
+        Returns the current state of the auth
+        ** Affects: None **
+    """
+    def get_auth(self, session=False):
+        session = self.session_focus if session is False else session
+        if session is True:
+            return self.session.auth
+        else:
+            return self.request_object.auth
+
+    """
+        Set a proxy that the request will use
+
+        @proxies
+            The proxy object as defined in the requests library
+    """
+    def set_proxy(self, proxies=None):
```

Comment:
I would separate out the connection configuration into a separate class and then pass that to the HttpHelper. This should also help us generify the actual yaml configuration and is better separation of concerns.